File: python/6 kyu/130.py
# https://www.codewars.com/kata/559536379512a64472000053/train/python

# Passed

def play_pass(s, n):
    result = str()
    n = n if n < 27 else n % 26
    for idx, char in enumerate(s):
        modified_char = char
        if char.isalpha():
            modified_char = modified_char.upper()
            code = ord(char) + n
            if code > 90:
                code -= 26 # circular shift

            modified_char = chr(code)
        elif char.isdigit():
            modified_char = str(9 - int(char))

        if idx % 2 == 0:
            modified_char = modified_char.upper()
        else:
            modified_char = modified_char.lower()

        result += modified_char

    result = result[::-1]
    return result    

output = play_pass("IN 2012 TWO CAMBRIDGE UNIVERSITY RESEARCHERS ANALYSED PASSPHRASES FROM THE AMAZON PAY SYSTEM...", 23)
print(output)

File: nodejs/7 kyu/130.js
// https://www.codewars.com/kata/559f89598c0d6c9b31000125/train/javascript

// Passed

function archersReady(archers) {
  return archers.length === 0
    ? false
    : archers.every((archer_arrow_count) => archer_arrow_count >= 5);
}

const output = archersReady([1, 2, 3, 4, 5, 6, 7, 8]);
console.log(output);